From sane-devel-admin@mostang.com  Sun Feb 24 20:04:53 2002
Return-Path: <sane-devel-admin@mostang.com>
Received: from localhost (localhost.localdomain [127.0.0.1])
	by meme.surrey.redhat.com (8.11.6/8.11.6) with ESMTP id g1OK4qb10110
	for <twaugh@localhost>; Sun, 24 Feb 2002 20:04:53 GMT
Received: from devserv.devel.redhat.com [172.16.58.1]
	by localhost with IMAP (fetchmail-5.9.0)
	for twaugh@localhost (single-drop); Sun, 24 Feb 2002 20:04:53 +0000 (GMT)
Received: from lacrosse.corp.redhat.com (IDENT:root@lacrosse.corp.redhat.com [172.16.52.154])
	by devserv.devel.redhat.com (8.11.6/8.11.0) with ESMTP id g1OE8qf15125;
	Sun, 24 Feb 2002 09:08:52 -0500
Received: from localhost.localdomain (int-mx1.corp.redhat.com [172.16.52.254])
	by lacrosse.corp.redhat.com (8.11.6/8.9.3) with ESMTP id g1OE8qs22685;
	Sun, 24 Feb 2002 09:08:52 -0500
Received: from mx1.redhat.com (mx1.redhat.com [172.16.48.31])
	by localhost.localdomain (8.11.6/8.11.6) with SMTP id g1OE8qA20694;
	Sun, 24 Feb 2002 09:08:52 -0500
Received: from panda.mostang.com (adsl-66-127-195-58.dsl.snfc21.pacbell.net [66.127.195.58])
	by mx1.redhat.com (8.11.6/8.11.6) with SMTP id g1OE8TQ05632;
	Sun, 24 Feb 2002 09:08:29 -0500
Received: from panda.mostang.com (localhost.localdomain [127.0.0.1])
	by panda.mostang.com (8.11.2/8.11.2) with ESMTP id g1OE6bE11246;
	Sun, 24 Feb 2002 06:06:37 -0800
Received: from mailout08.sul.t-online.com (mailout08.sul.t-online.com [194.25.134.20])
	by panda.mostang.com (8.11.2/8.11.2) with ESMTP id g1OE5iE11230
	for <sane-devel@mostang.com>; Sun, 24 Feb 2002 06:05:45 -0800
Received: from fwd08.sul.t-online.de 
	by mailout08.sul.t-online.com with smtp 
	id 16eyqK-0007o1-0D; Sun, 24 Feb 2002 14:37:32 +0100
Received: from vortex.meier-geinitz.de (520024250054-0001@[217.225.206.91]) by fmrl08.sul.t-online.com
	with esmtp id 16eyqF-0puwVcC; Sun, 24 Feb 2002 14:37:27 +0100
Received: from hmg1.meier-geinitz.de ([192.168.0.1] helo=hmg1)
	by vortex.meier-geinitz.de with esmtp (Exim 3.34 #1 (Debian))
	id 16eyrD-0004AA-00
	for <sane-devel@mostang.com>; Sun, 24 Feb 2002 14:38:27 +0100
Received: from hmg by hmg1 with local (Exim 3.33 #1 (Debian))
	id 16eyrC-0006V2-00
	for <sane-devel@mostang.com>; Sun, 24 Feb 2002 14:38:26 +0100
From: Henning Meier-Geinitz <henning@meier-geinitz.de>
To: sane-devel@mostang.com
Subject: Re: [sane-devel] 'scanimage -L' fails first time with modular SCSI drivers
Message-ID: <20020224133825.GB1685@vortex.meier-geinitz.de>
Mail-Followup-To: sane-devel@mostang.com
References: <20020219144024.I3029@redhat.com> <20020219184312.GG6922@vortex.meier-geinitz.de> <20020219190520.GI6922@vortex.meier-geinitz.de> <20020222140214.B394@redhat.com>
Mime-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
In-Reply-To: <20020222140214.B394@redhat.com>
User-Agent: Mutt/1.3.27i
X-Sender: 520024250054-0001@t-dialin.net
Sender: sane-devel-admin@mostang.com
Errors-To: sane-devel-admin@mostang.com
X-BeenThere: sane-devel@www.mostang.com
X-Mailman-Version: 2.0.8
Precedence: bulk
List-Help: <mailto:sane-devel-request@www.mostang.com?subject=help>
List-Post: <mailto:sane-devel@www.mostang.com>
List-Subscribe: <http://www.mostang.com/mailman/listinfo/sane-devel>,
	<mailto:sane-devel-request@www.mostang.com?subject=subscribe>
List-Id: SANE developer's mailing list. <sane-devel.www.mostang.com>
List-Unsubscribe: <http://www.mostang.com/mailman/listinfo/sane-devel>,
	<mailto:sane-devel-request@www.mostang.com?subject=unsubscribe>
List-Archive: <http://www.mostang.com/pipermail/sane-devel/>
Date: Sun, 24 Feb 2002 14:38:25 +0100
Status: RO
Content-Length: 1457
Lines: 51

Hi,

On Fri, Feb 22, 2002 at 02:02:14PM +0000, Tim Waugh wrote:
> Could we add some hook to try to get the modules loaded if
> /proc/scsi/scsi doesn't exist?  Just opening a generic SCSI device is
> enough to load them.

Is anybody against the following patch?

Bye,
  Henning

Index: sanei/sanei_scsi.c
===================================================================
RCS file: /cvsroot/external/sane/sane-backends/sanei/sanei_scsi.c,v
retrieving revision 1.31
diff -u -u -r1.31 sanei_scsi.c
--- sanei_scsi.c	2002/02/09 16:37:21	1.31
+++ sanei_scsi.c	2002/02/24 13:35:40
@@ -2440,11 +2440,13 @@
   #define FOUND_LUN     128
   #define FOUND_ALL     255
 
+  #define FIND_DEVICES_TOUCH_FILE "/dev/sg0"
+
   size_t findvendor_len = 0, findmodel_len = 0, findtype_len = 0;
   char vendor[32], model[32], type[32], revision[32];
   int bus, channel, id, lun;
 
-  int number, i, j, definedd;
+  int number, i, j, definedd, fd;
   char line[256], dev_name[128], *c1, *c2, ctmp;
   const char *string;
   FILE *proc_fp;
@@ -2484,6 +2486,12 @@
   param[7].u.i = &lun;
 
   DBG_INIT ();
+
+  /* Make sure that the kernel SCSI drivers are loaded before reading 
+	 PROCFILE */
+  fd = open (FIND_DEVICES_TOUCH_FILE, O_RDONLY);
+  if (fd >= 0)
+	close (fd);
 
   proc_fp = fopen (PROCFILE, "r");
   if (!proc_fp)
_______________________________________________
Sane-devel mailing list
Sane-devel@www.mostang.com
http://www.mostang.com/mailman/listinfo/sane-devel
